dapr/proto/runtime/v1/conversation.proto

```diff
@@ -0,0 +1,342 @@
+
+import "google/protobuf/any.proto";
+import "google/protobuf/wrappers.proto";
+
+option go_package = "github.com/dapr/dapr/pkg/proto/runtime/v1;runtime";
+
+// ConversationRequest is the request object for Conversation.
+message ConversationRequest {
+  option deprecated = true;
+  
+  // The name of Conversation component
+  string name = 1;
+
+  // The ID of an existing chat (like in ChatGPT)
+  optional string contextID = 2;
+
+  // Inputs for the conversation, support multiple input in one time.
+  repeated ConversationInput inputs = 3;
+
+  // Parameters for all custom fields.
+  map<string, google.protobuf.Any> parameters = 4;
+
+  // The metadata passing to conversation components.
+  map<string, string> metadata = 5;
+
+  // Scrub PII data that comes back from the LLM
+  optional bool scrubPII = 6;
+
+  // Temperature for the LLM to optimize for creativity or predictability
+  optional double temperature = 7;
+}
+
+// ConversationRequestV1Alpha2 is the new request object for Conversation.
+// Many of these fields are inspired by openai.ChatCompletionNewParams
+// https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2106
+message ConversationRequestV1Alpha2 {
+  // The name of Conversation component
+  string name = 1;
+
+  // The ID of an existing chat (like in ChatGPT)
+  optional string context_id = 2;
+
+  // Inputs for the conversation, support multiple input in one time.
+  // This is the revamped conversation inputs better matching openai.
+  repeated ConversationInputV1Alpha2 inputs = 3;
+
+  // Parameters for all custom fields.
+  map<string, google.protobuf.Any> parameters = 4;
+
+  // The metadata passing to conversation components.
+  map<string, string> metadata = 5;
+
+  // Scrub PII data that comes back from the LLM
+  optional bool scrub_pii = 6;
+
+  // Temperature for the LLM to optimize for creativity or predictability
+  optional double temperature = 7;
+
+  // Tools register the tools available to be used by the LLM during the conversation.
+  // These are sent on a per request basis.
+  // The tools available during the first round of the conversation
+  // may be different than tools specified later on.
+  repeated ConversationTools tools = 8;
+
+  // Controls which (if any) tool is called by the model. 
+  // `none` means the model will not call any tool and instead generates a message. 
+  // `auto` means the model can pick between generating a message or calling one or more tools. 
+  // `required` means the model must call one or more tools. 
+  // Specifying a particular tool via:
+  // `{"type": "function", "function": {"name": "my_function"}}` forces the model to call that tool.
+  // `none` is the default when no tools are present.
+  // `auto` is the default if tools are present.
+  // ref: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L1976
+  // ref: https://python.langchain.com/docs/how_to/tool_choice/
+  optional google.protobuf.StringValue tool_choice = 9;
```

Comment:
Why do not use string directly here?

Reply:
I agree we should be using a `string` directly here. `optional string` achieves the same behaviour you are describing.